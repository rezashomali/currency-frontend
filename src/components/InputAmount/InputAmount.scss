.inputAmount {
  .inputAmount__input {
    font-size: 30px;
  }
  .inputAmount__symbol p {
    font-size: 25px;
  }

  .inputAmount__error {
    color: #e43131;
    font-size: 13px;
    position: absolute;
    bottom: -20px;
  }
}
