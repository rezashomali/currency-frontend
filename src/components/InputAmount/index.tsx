import React from "react";
import {
  FormControl,
  InputLabel,
  Input,
  InputAdornment,
} from "@material-ui/core";
import "./InputAmount.scss";

interface Props {
  onAmountChange: any;
  symbol: string;
}

const InputAmount: React.FC<Props> = ({ onAmountChange, symbol }) => {
  return (
    <FormControl fullWidth className="inputAmount">
      <InputLabel htmlFor="standard-adornment-amount">Amount</InputLabel>
      <Input
        className="inputAmount__input"
        // id="standard-adornment-amount"
        // value={values.amount}
        onChange={onAmountChange}
        startAdornment={
          <InputAdornment className="inputAmount__symbol" position="start">
            {symbol}
          </InputAdornment>
        }
      />
    </FormControl>

    // <FormControl fullWidth>
    //   <InputLabel htmlFor="outlined-adornment-amount">Amount</InputLabel>
    //   <OutlinedInput
    //     onChange={onAmountChange}
    //     startAdornment={
    //       <InputAdornment position="start">{symbol}</InputAdornment>
    //     }
    //     labelWidth={60}
    //   />
    // </FormControl>
  );
};

export default InputAmount;
