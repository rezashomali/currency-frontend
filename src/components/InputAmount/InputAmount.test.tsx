import React from "react";
import renderer from "react-test-renderer";
import InputAmount from "./index";

test("InputAmount snapshot ", () => {
  const component = renderer.create(
    <InputAmount amount={0} symbol={"$"} onChange={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
