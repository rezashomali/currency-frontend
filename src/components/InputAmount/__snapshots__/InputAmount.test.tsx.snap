// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAmount snapshot  1`] = `
<div
  className="MuiFormControl-root inputAmount MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="standard-adornment-amount"
  >
    Amount
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline inputAmount__input MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root inputAmount__symbol MuiInputAdornment-positionStart"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        $
      </p>
    </div>
    <input
      aria-invalid={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="number"
      value={0}
    />
  </div>
  
</div>
`;
